src/java/org/apache/cassandra/service/accord/AccordKeyspace.java

```diff
@@ -115,12 +115,12 @@
     private static final Logger logger = LoggerFactory.getLogger(AccordKeyspace.class);
 
     public static final String COMMANDS = "commands";
-    public static final String COMMAND_SERIES = "command_series";
     public static final String COMMANDS_FOR_KEY = "commands_for_key";
 
     private static final String TIMESTAMP_TUPLE = "tuple<bigint, bigint, int, bigint>";
     private static final TupleType TIMESTAMP_TYPE = new TupleType(Lists.newArrayList(LongType.instance, LongType.instance, Int32Type.instance, LongType.instance));
     private static final String KEY_TUPLE = "tuple<uuid, blob>";
+    private static final int CURRENT_VERSION = 1;
```

Comment:
why?  The point was to decouple the storage version from the messaging version; if messaging bumps that doesn't mean storage did

Reply:
yeah good point... the serialized used in the end also work on network, so versions *has* to be network in order to avoid duplication...

>  I'm already doing this in AccordPartialCommand.PartialCommandSerializer.Version. Why not promote that enum to AccordSerializerVersion and use it instead?

so the current value is `VERSION_0(0, MessagingService.current_version);` which has the bug that we redefine if anyone bumps; so should become `VERSION_0(0, MessagingService.VERSION_40);`...

Now, lets say we do promote, then we would need to have something like an AccordSerializer which is a `IVersionedSerializer` but that maps `int version` to `enum AccordSerializerVersion` and delegates to copies of the methods that take the `AccordSerializerVersion`...

One thing also being pointed out here is that `PartialCommandSerializer` and `LocalVersionedSerializer` duplicate logic, so we should unify this.

Ill take a stab at unifying this logic...

Reply:
ok push new commit 39c27064f085371718495e6fa557ef0ae3ff3f14 which attempts to unify the 2 code paths